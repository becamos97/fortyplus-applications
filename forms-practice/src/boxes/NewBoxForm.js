import React, { useState } from "react";

const INITIAL = { width: "100", height: "100", backgroundColor: "#ff0000" };

export default function NewBoxForm({ addBox }) {
  const [formData, setFormData] = useState(INITIAL);

  function handleChange(e) {
    const { name, value } = e.target;
    setFormData(f => ({ ...f, [name]: value }));
  }

  function handleSubmit(e) {
    e.preventDefault();
    addBox(formData);
    setFormData(INITIAL);
  }

  return (
    <form onSubmit={handleSubmit} aria-label="new-box-form">
      <label htmlFor="width">Width (px):</label>
      <input id="width" name="width" value={formData.width} onChange={handleChange} />

      <label htmlFor="height" style={{ marginLeft: 8 }}>Height (px):</label>
      <input id="height" name="height" value={formData.height} onChange={handleChange} />

      <label htmlFor="backgroundColor" style={{ marginLeft: 8 }}>Color:</label>
      <input id="backgroundColor" name="backgroundColor" value={formData.backgroundColor} onChange={handleChange} />

      <button style={{ marginLeft: 8 }}>Add Box</button>
    </form>
  );
}