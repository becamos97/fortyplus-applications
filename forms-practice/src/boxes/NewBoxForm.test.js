import React from "react";
import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";
import "@testing-library/jest-dom";

it("renders without crashing", () => {
  render(<NewBoxForm addBox={() => {}} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewBoxForm addBox={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});