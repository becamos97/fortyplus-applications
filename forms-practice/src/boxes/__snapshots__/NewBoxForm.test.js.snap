// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    aria-label="new-box-form"
  >
    <label
      for="width"
    >
      Width (px):
    </label>
    <input
      id="width"
      name="width"
      value="100"
    />
    <label
      for="height"
      style="margin-left: 8px;"
    >
      Height (px):
    </label>
    <input
      id="height"
      name="height"
      value="100"
    />
    <label
      for="backgroundColor"
      style="margin-left: 8px;"
    >
      Color:
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      value="#ff0000"
    />
    <button
      style="margin-left: 8px;"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
